sync/src/types.rs

```diff
@@ -1167,15 +1178,30 @@ impl SyncShared {
         self.state.may_set_shared_best_header(header_view);
     }
 
-    pub fn get_header_view(&self, hash: &Byte32) -> Option<HeaderView> {
+    pub fn get_header_view(
+        &self,
+        hash: &Byte32,
+        store_first_opt: Option<bool>,
```

Comment:
is it possible to use `BlockStatus` here?

```
BlockStatus::UNKNOWN => None,
BlockStatus::STORED => Some(true),
others => Some(false),
```

Reply:
This assume that `tip.number() >= header.number()` => `BlockStatus::STORED`, I'm not sure about that.
I just search store first when `tip.number() >= header.number()`, but not mean `BlockStatus::STORED`.